packages/calcite-components/src/components/text-area/text-area.tsx

```diff
@@ -426,34 +430,61 @@ export class TextArea
     };
   }
 
-  private replacePlaceholdersInMessages(): string {
-    return this.messages.tooLong
-      .replace("{maxLength}", this.localizedCharacterLengthObj.maxLength)
-      .replace("{currentLength}", this.localizedCharacterLengthObj.currentLength);
+  private replacePlaceholders(
+    template: string,
+    placeholderToValueObject: { [key: string]: string },
+  ): string {
+    let result;
+    for (const key in placeholderToValueObject) {
+      result = template.replace(key, placeholderToValueObject[key]);
+    }
+    return result;
+  }
+
+  private replacePlaceholdersInLongMessages(): string {
+    return this.replacePlaceholders(this.messages.tooLong, {
+      "{currentLength}": this.localizedCharacterLengthObj.currentLength,
+      "{maxLength}": this.localizedCharacterLengthObj.maxLength,
+    });
   }
 
-  private isCharacterLimitExceeded(): boolean {
+  private replacePlaceholdersInShortMessages(): string {
```

Comment:
nitpick: can we use a different name here to be more meaningful.

replaceMinLengthPlaceHolderMessage( ) or getMinLengthMessage( ).